### PR TITLE
Added search function to search movies by name

### PR DESCRIPTION
**Pull request by Nusrath-Jahan**

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -3,13 +3,35 @@
...
-
 <body>
-   
-    <script src="script.js"></script>
-    <script src="arrayList.js"></script>
+    <header>
+        <p class="site-name"><i class="fa fa-video-camera col_red me-1" style="margin-right: 10px"></i>Flimy Fly</p>
+        <div class="forUser">
+        <span>
+        <input type="search" id="keyword-search" placeholder="search" required style="width: 300px; height: 40px;"/>
```
Comment (vdaguenet) [inline index.html]: Adding style on the HTML element is usually a bad practice. Prefer using CSS files for adding styling

File: script.js [hunk 2]
```diff
@@ -1,2 +1,168 @@
...
+}
+createMoviesPage();
+
+// JS 2
+const searchBtn = document.getElementById("search-icon");
+
+// function to search
+const inputField = document.getElementById("keyword-search");
+searchBtn.addEventListener("click", () => {
+  const inputStr = inputField.value.toLowerCase();
```
Comment (vdaguenet) [inline script.js]: Don't forget to also `trim` the input value to avoid searching for spaces.

File: script.js [hunk 3]
```diff
@@ -1,2 +1,168 @@
...
+  nowStreaming.innerHTML = "";
+  let found = false; // Flag to check if any movie is found
+
+  movies.forEach((movie) => {
```
Comment (vdaguenet) [inline script.js]: A `forEach` plus a `found` boolean can work fine. But you can probably simplify your code with one of the methods available on arrays. Have a look at https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array#instance_methods. I am sure you will find something that fits your needs.

File: script.js [hunk 4]
```diff
@@ -1,2 +1,168 @@
...
+    if (
+      movie.title.toLowerCase().includes(inputStr) ||
+      movie.description.toLowerCase().includes(inputStr)
+    ) {
+      found = true;
+      const searchMovieElement = createMovieCard(movie);
+      nowStreaming.appendChild(searchMovieElement);
+    }
+    if (!found) {
+      nowStreaming.innerHTML = `<p class="not-found-message">Not Found</p>`;
```
Comment (vdaguenet) [inline script.js]: This message is visible even though there is a result:
image

File: styles.css [hunk 5]
```diff
@@ -1,3 +1,138 @@
...
+    margin-top: 10px;
+    color: rgb(68, 68, 109);
+  }
+  .not-found-message {
+    color: red; 
+    font-size: 50px;
+  }
+
+  
+  Fading animation
```
Comment (vdaguenet) [inline styles.css]: I guess this should be a comment

File: styles.css [hunk 6]
```diff
@@ -1,3 +1,138 @@
...
+  display: flex;
+  justify-content: space-between;
+  align-content: center;
+  margin: 20px 0 20px 10px;
+  padding: 20px 0 0 50px;
+}
+.main{
+    margin: 20px 0 20px 50px;
+}
+  .site-name {
```
Comment (vdaguenet) [inline styles.css]:
```suggestion
.site-name {
```
Be careful with the indentation.  From this line, everything is one tab too much to the right.

**Review by vdaguenet — CHANGES_REQUESTED** [review]
The brief for the week 2 includes `a function for sorting the movie array by one or a few selected properties.`. It does not have to react to user inputs but a sort function must be implemented.

File: styles.css (hunk 6 above)
Comment (Nusrath-Jahan) [reply styles.css]: hello @vdaguenet , 

Thank you for your valuable review and suggestions.
I have implemented 2 includes a function for sorting the movie array by one or a few selected properties. Now It  is  react to user inputs .

Thank you

**Review by vdaguenet — APPROVED** [review]
Very cool!

**Pull request merged**